hs_core/hydroshare/resource.py

```diff
@@ -686,7 +690,18 @@ def delete_resource(pk):
 
     Note:  Only HydroShare administrators will be able to delete formally published resour
     """
-    utils.get_resource_by_shortkey(pk).delete()
+    #utils.get_resource_by_shortkey(pk).delete()
+    res = utils.get_resource_by_shortkey(pk)
+    for fl in res.files.all():
+        fl.resource_file.delete()
+
+    for bag in res.bags.all():
+        bag.bag.delete()
+        bag.delete()
+
+    res.metadata.delete_all_elements()
+    # deleting the metadata object deletes the resource also
+    res.metadata.delete()
```

Comment:
If we just delete the resource, the associated resource files and the bag do not get deleted. Django documentation says it should delete theses generic related objects. I am not sure why the cascade delete is not working for us.